### PR TITLE
add Alchemistry compat

### PR DESCRIPTION
**Pull request by WaitingIdly**
what this PR adds:
- Adds compat for the `atomizer`, `combiner`, `dissolver`, `electrolyzer`, `evaporator`, `liquifier`.
- i18n wiki keys for the above
- examples file for the above
- `element` game object handler to get the alchemistry element or compound for the given name

something that is not added in this PR:
- crafttweaker, through native compat, adds Create Element and Create Compound helpers. these are not added in this PR. a TODO is added for these helpers.

note: when testing, i suggest also enabling Draconic Evolution for the Creative Power Source (no way to get power otherwise)

### CODE REVIEW

File: src/main/java/com/cleanroommc/groovyscript/compat/mods/alchemistry/Atomizer.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    public boolean remove(AtomizerRecipe recipe) {
+        if (ModRecipes.INSTANCE.getAtomizerRecipes().removeIf(r -> r == recipe)) {
+            addBackup(recipe);
+            return true;
+        }
+        return false;
+    }
+
+    @MethodDescription(description = "groovyscript.wiki.removeByOutput", example = @Example(value = "item('alchemistry:compound:7')", commented = true))
+    public boolean removeByOutput(ItemStack output) {
```
Comment (brachy84) [inline src/main/java/com/cleanroommc/groovyscript/compat/mods/alchemistry/Atomizer.java:52]: Rather use `IIngredient` and `.test()`. This applies to other machines for input and output as well.

File: src/main/java/com/cleanroommc/groovyscript/compat/mods/alchemistry/Atomizer.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                return true;
+            }
+            return false;
+        });
+    }
+
+    @MethodDescription(description = "groovyscript.wiki.removeByInput")
+    public boolean removeByInput(IIngredient input) {
+        return removeByInput(IngredientHelper.toFluidStack(input));
+    }
```
Comment (brachy84) [inline src/main/java/com/cleanroommc/groovyscript/compat/mods/alchemistry/Atomizer.java:76]: You can't just use `IIngredient` and then cast to `FluidStack` 💀

File: src/main/java/com/cleanroommc/groovyscript/compat/mods/alchemistry/Electrolyzer.java [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+                }
+            }
+            return false;
+        });
+    }
+
+    @MethodDescription(description = "groovyscript.wiki.removeByInput")
+    public boolean removeByInput(IIngredient input) {
+        return removeByInput(IngredientHelper.toItemStack(input));
+    }
```
Comment (brachy84) [inline src/main/java/com/cleanroommc/groovyscript/compat/mods/alchemistry/Electrolyzer.java:95]: 💀

**Review by brachy84 — CHANGES_REQUESTED** [review]
Please add simple add methods for those who dont like builders

File: src/main/java/com/cleanroommc/groovyscript/compat/mods/alchemistry/Atomizer.java (hunk 2 above)
Comment (WaitingIdly) [reply src/main/java/com/cleanroommc/groovyscript/compat/mods/alchemistry/Atomizer.java]: :skull:

File: src/main/java/com/cleanroommc/groovyscript/compat/mods/alchemistry/Atomizer.java (hunk 1 above)
Comment (WaitingIdly) [reply src/main/java/com/cleanroommc/groovyscript/compat/mods/alchemistry/Atomizer.java]: alright, changed them. there are a few parts of older compat that will also need this change. uh, most really.

**Review by brachy84 — APPROVED** [review]

**Pull request merged**